app/controllers/accounts_controller.rb

```diff
@@ -0,0 +1,67 @@
+class AccountsController < ApplicationController
+  layout 'onboarding_application'
+  include Steps::StepsList
+
+  before_action :authenticate_user!
+
+  def add_plan
+    if selected_plan
+      update_account_and_redirect(
+        select_addons_onboarding_path(onboarding),
+        select_plan_onboarding_path(onboarding),
+        3
+      )
+    else
+      flash[:alert] = 'Plan is required'
+      redirect_to select_plan_onboarding_path(onboarding)
+    end
+  end
+
+  private
+
+  def onboarding
+    @onboarding ||= Onboarding.find(params[:id])
+  end
+
+  def account
+    @account ||= onboarding.account
+  end
+
+  def selected_payment_recurrence
+    @selected_payment_recurrence ||= PaymentRecurrence.find_by(
+      name: permitted_params[:plan][:payment_recurrence][:name]
+    )
+  end
+
+  def selected_plan
+    return account.plan unless permitted_params[:plan][:name]
+
+    plan_name = permitted_params[:plan][:name].split('_')[0]
+    @selected_plan ||= Plan.where(name: plan_name).find_by(payment_recurrence: selected_payment_recurrence)
```

Comment:
Just insert both conditions inside the `find_by`. Using `where` separately doesn't add any value
```suggestion
    @selected_plan ||= Plan.find_by(payment_recurrence: selected_payment_recurrence, name: plan_name)
```